src/Adapter/MSTest.CoreAdapter/Extensions/TestCaseExtensions.cs

```diff
@@ -24,8 +24,10 @@ internal static UnitTestElement ToUnitTestElement(this TestCase testCase, string
             var testClassName = testCase.GetPropertyValue(Constants.TestClassNameProperty) as string;
             var declaringClassName = testCase.GetPropertyValue(Constants.DeclaringClassNameProperty) as string;
 
-            var parts = testCase.FullyQualifiedName.Split('.');
-            var name = parts[parts.Length - 1];
+            var fullyQualifiedName = testCase.FullyQualifiedName;
+
+            // not using Replace because there can be multiple instances of that string
+            var name = fullyQualifiedName.StartsWith($"{testClassName}.") ? fullyQualifiedName.Remove(0, $"{testClassName}.".Length) : fullyQualifiedName;
```

Comment:
nit: Maybe split the line to make it easier to read ?

`fullyQualifiedName.StartsWith($"{testClassName}.")`
`    ? fullyQualifiedName.Remove(0, $"{testClassName}.".Length)`
`    : fullyQualifiedName;`

Reply:
Next time. :)

Reply:
done